### PR TITLE
Adding logic and configuration for test coverage

### PR DESCRIPTION
**Pull request by traviswestbrook**

### CODE REVIEW

File: fork-runner/src/main/java/com/shazam/fork/runner/listeners/CoverageListener.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    public void testEnded(TestIdentifier test, Map<String, String> testMetrics) {
+    }
+
+    @Override
+    public void testRunFailed(String errorMessage) {
+    }
+
+    @Override
+    public void testRunStopped(long elapsedTime) {
+    }
```
Comment (ndtaylor) [inline fork-runner/src/main/java/com/shazam/fork/runner/listeners/CoverageListener.java:65]: do nothing comments?

File: fork-runner/src/main/java/com/shazam/fork/runner/listeners/CoverageListener.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    @Override
+    public void testRunEnded(long elapsedTime, Map<String, String> runMetrics) {
+        try {
+            final String remoteFile = RemoteFileManager.getCoverageFileName(testClass.getName());
```
Comment (ndtaylor) [inline fork-runner/src/main/java/com/shazam/fork/runner/listeners/CoverageListener.java]: Move this out of the try block.

File: fork-runner/src/main/java/com/shazam/fork/system/io/FileManager.java [hunk 3]
```diff
@@ -1,23 +1,31 @@
...
+ * the specific language governing permissions and limitations under the License.
  */
 package com.shazam.fork.system.io;
 
 import com.android.ddmlib.testrunner.TestIdentifier;
-import com.shazam.fork.model.*;
-
-import org.apache.commons.io.filefilter.*;
-
-import java.io.*;
```
Comment (ndtaylor) [inline fork-runner/src/main/java/com/shazam/fork/system/io/FileManager.java:17]: revert

File: fork-runner/src/main/java/com/shazam/fork/runner/listeners/TestRunListenersFactory.java [hunk 4]
```diff
@@ -13,7 +13,9 @@
 import com.android.ddmlib.testrunner.ITestRunListener;
 import com.google.gson.Gson;
 import com.shazam.fork.Configuration;
-import com.shazam.fork.model.*;
```
Comment (ndtaylor) [inline fork-runner/src/main/java/com/shazam/fork/runner/listeners/TestRunListenersFactory.java]: revert

**Comment by ndtaylor** [comment]
lgtm

File: fork-runner/src/main/java/com/shazam/fork/runner/listeners/CoverageListener.java (hunk 1 above)
Comment (traviswestbrook) [inline fork-runner/src/main/java/com/shazam/fork/runner/listeners/CoverageListener.java:65]: > do nothing comments?

That doesn't appear to be their convention.

File: fork-runner/src/main/java/com/shazam/fork/runner/listeners/CoverageListener.java (hunk 2 above)
Comment (traviswestbrook) [inline fork-runner/src/main/java/com/shazam/fork/runner/listeners/CoverageListener.java]: > Move this out of the try block.

done

File: fork-runner/src/main/java/com/shazam/fork/system/io/FileManager.java (hunk 3 above)
Comment (traviswestbrook) [inline fork-runner/src/main/java/com/shazam/fork/system/io/FileManager.java:17]: > revert

done

File: fork-runner/src/main/java/com/shazam/fork/runner/listeners/TestRunListenersFactory.java (hunk 4 above)
Comment (traviswestbrook) [inline fork-runner/src/main/java/com/shazam/fork/runner/listeners/TestRunListenersFactory.java]: > revert

done

**Comment by KennethNickles** [comment]
ship it

**Pull request merged**